C7/Game.cs

```diff
@@ -436,108 +404,12 @@ public override void _UnhandledInput(InputEvent @event)
-					default: return; // Impossible
-					}
-					new MsgMoveUnit(CurrentlySelectedUnit.guid, dir).send();
-					setSelectedUnit(CurrentlySelectedUnit);	//also triggers updating the lower-left info box
-				}
-			}
-			else if ((eventKeyDown.Keycode >= Godot.Key.Home) && (eventKeyDown.Keycode <= Godot.Key.Pagedown))
-			{ // Move units with the arrow and fn keys
-				if (CurrentlySelectedUnit != MapUnit.NONE)
-				{
-					TileDirection dir;
-					switch (eventKeyDown.Keycode) {
-					case Godot.Key.Home:     dir = TileDirection.NORTHWEST; break; // fn-left arrow
-					case Godot.Key.End:      dir = TileDirection.SOUTHWEST; break; // fn-right arrow
-					case Godot.Key.Left:     dir = TileDirection.WEST;      break;
-					case Godot.Key.Up:       dir = TileDirection.NORTH;     break;
-					case Godot.Key.Right:    dir = TileDirection.EAST;      break;
-					case Godot.Key.Down:     dir = TileDirection.SOUTH;     break;
-					case Godot.Key.Pageup:   dir = TileDirection.NORTHEAST; break; // fn-up arrow
-					case Godot.Key.Pagedown: dir = TileDirection.SOUTHEAST; break; // fn-down arrow
-					default: return; // Impossible
-					}
-					new MsgMoveUnit(CurrentlySelectedUnit.guid, dir).send();
-					setSelectedUnit(CurrentlySelectedUnit);	//also triggers updating the lower-left info box
-				}
-			}
-			else if (eventKeyDown.Keycode == Godot.Key.G && eventKeyDown.IsCommandOrControlPressed())
-			{
-				mapView.gridLayer.visible = !mapView.gridLayer.visible;
-			}
-			else if (eventKeyDown.Keycode == Godot.Key.Escape)
-			{
-				if (!inUnitGoToMode) {
-					log.Debug("Got request for escape/quit");
-					PopupOverlay popupOverlay = GetNode<PopupOverlay>(PopupOverlay.NodePath);
-					popupOverlay.ShowPopup(new EscapeQuitPopup(), PopupOverlay.PopupCategory.Info);
-				}
-			}
-			else if (eventKeyDown.Keycode == Godot.Key.Z)
-			{
-				if (mapView.cameraZoom != 1) {
-					mapView.setCameraZoomFromMiddle(1.0f);
-					VSlider slider = GetNode<VSlider>("CanvasLayer/SlideOutBar/VBoxContainer/Zoom");
-					slider.Value = 1.0f;
-				}
-				else {
-					mapView.setCameraZoomFromMiddle(0.5f);
-					VSlider slider = GetNode<VSlider>("CanvasLayer/SlideOutBar/VBoxContainer/Zoom");
-					slider.Value = 0.5f;
-				}
-			}
-			else if (eventKeyDown.Keycode == Godot.Key.Shift && ! eventKeyDown.Echo)
-			{
-				SetAnimationsEnabled(false);
-			}
-			else if (eventKeyDown.Keycode == Godot.Key.O && eventKeyDown.ShiftPressed && eventKeyDown.IsCommandOrControlPressed() && eventKeyDown.AltPressed) {
+		} else if (@event is InputEventMouseMotion eventMouseMotion && IsMovingCamera) {
+			GetViewport().SetInputAsHandled();
+			mapView.cameraLocation += OldPosition - eventMouseMotion.Position;
+			OldPosition = eventMouseMotion.Position;
+		} else if (@event is InputEventKey eventKeyDown && eventKeyDown.Pressed) {
+			if (eventKeyDown.Keycode == Godot.Key.O && eventKeyDown.ShiftPressed && eventKeyDown.IsCommandOrControlPressed() && eventKeyDown.AltPressed) {
```

Comment:
I don't yet understand how this API works but here we have a few hardcoded Keycodes and elsewhere there are references to C7Action. Can these be replaced with new actions?

Reply:
updated all the string literals with C7Action string constants. I didn't add an action for this since it's a debug feature for development